case-lib/lib.sh

```diff
@@ -682,26 +682,11 @@ is_zephyr()
 
 is_firmware_file_zephyr()
 {
-    local firmware_name firmware_path znum platform
-    # get the FW name and path
-    # TODO: optimize this part after driver can expose the FW path to userspace via debugfs
-    if is_ipc4; then
-        firmware_name=dsp_basefw.bin
-        fw_mod_para=/sys/module/snd_sof_pci/parameters/fw_path
-        if [ -s "$fw_mod_para" ]; then
-            firmware_path=$(cat $fw_mod_para)
-        else
-            # # FIXME: the kernel driver should give us the FW path
-            # https://github.com/thesofproject/linux/issues/3867
-            die "Failed to get the IPC4 FW path."
-        fi
-    else # for IPC3
-        platform=$(sof-dump-status.py -p)
-        firmware_name=sof-$platform.ri
-        firmware_path=$(sof-dump-status.py -P)
-    fi
+    local firmware_path znum
 
-    znum=$(strings "/lib/firmware/$firmware_path/$firmware_name" | grep -c -i zephyr)
+    firmware_path=$(journalctl_cmd | grep "request_firmware" | tail -n 1 |
+                    awk -F ' ' '{print $(NF-1)}')
```

Comment:
If you check the kernel source code, you will find there are some drivers which will dump "request_firmware" string. For example, ./drivers/net/wireless/intel/ipw2x00/ipw2200.c or ./drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c. This means there is potential conflict if we use the dmesg to get the firmware path. 

This should work on our CI machines currently. But customers outside intel may find it doesn't work. Or our later platform using different hw configuration will have issues to use dmesg.